### PR TITLE
[bytecode verifier] Fix stack management in Reference Safety

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Fixed stack balancing issues in the new borrow checker implementation
- @wrwg This will fix your bug

## Test Plan

- Ran em
- new test
- verified problematic code was fixed

### CODE REVIEW

File: language/bytecode-verifier/src/reference_safety/mod.rs [hunk 1]
```diff
@@ -204,28 +204,34 @@ fn execute_inner(
         }
 
         Bytecode::MutBorrowGlobal(idx) => {
+            checked_verify!(verifier.stack.pop().unwrap().is_value());
```
Comment (tnowacki) [inline language/bytecode-verifier/src/reference_safety/mod.rs:207]: These are the actual bugs, the rest is debugging cleanup

**Comment by tnowacki** [comment]
@dariorussi r=darioncassel

**Comment by wrwg** [comment]
Thanks for the fast fix!

**Pull request closed without merging**